### ISSUE TITLE
Javascript error when trying to merge model with null value for relation

### ISSUE THREAD

**Issue by davincho** [js-data/js-data#215]
Code that reproduces the error: http://plnkr.co/edit/aJb2dKSVG7JE4CNucmYy?p=preview

We have two simple models:

```javascript
var Prod = store.defineResource({
    name: 'prod',
    relations: {
      hasOne: {
        comment: {
          localField: 'comment',
          foreignKey: 'prod_id'
        }
      }
    }
  });

  var Comment = store.defineResource({
    name: 'comment'
  });
```

Let's inject new prod object with nested comment object:

```javascript
Prod.inject({
    id: 1,
    comment: {
      id: 3,
      prod_id: 1
    }
  });
```

Now inject a new prod object with the same id (=merge as specified in http://www.js-data.io/docs/dsdefaults#onconflict) and a null value for ``comment``:

```javascript
Prod.inject({
     id: 1,
     name: 'test',
     comment: null
  });
```

**Result**: Javascript error in https://github.com/js-data/js-data/blob/master/dist/js-data.js#L2912 as ``obj`` is ``null``.

![shot](https://cloud.githubusercontent.com/assets/2635143/9983021/51b80e56-5ff0-11e5-9300-a0291f03df0d.png)

I would suggest to skip null values when being merged.

**Comment by davincho** [comment]
@jmdobry thx a lot, you are awesome!

**Comment by jmdobry** [comment]
The fix will be available when 2.4.0 is released.
